source/dataman/AMScanAxis.cpp

```diff
@@ -369,3 +370,49 @@ QString AMScanAxis::toString(const QString &units) const
 
 	return string;
 }
+
+double AMScanAxis::minimumRegionTime() const
+{
+	double minTime = 0;
+
+	// Iterate through the regions, identify the smallest region time.
+
+	for (int i = 0, count = regionCount(); i < count; i++) {
+
+		double regionDwell = regionAt(i)->regionTime();
```

Comment:
I believe the time needs to be casted.  `double regionDwell = double(regionAt(i)->regionTime());`

Also, I think the conditional needs to be a bit more strict:

```c++
if (minTime > regionDwell && regionDwell >= 0)  // This ensures that the times can't be negative if the region hasn't been fully specified.
    minTime = regionDwell;
```

Finally, the `minTime` needs to be initialized to positive infinite or the first region's time because this function will always return 0 unless the time is negative.